File: re-ssim-service/src/main/java/com/shinonometn/re/ssim/service/courses/plugin/TermMeta.java
package com.shinonometn.re.ssim.service.courses.plugin;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class TermMeta {
    private Integer courseCount;
    private String version;

    public TermMeta() {
    }

    public TermMeta(Integer courseCount) {
        this.courseCount = courseCount;
    }

    public Integer getCourseCount() {
        return courseCount;
    }

    public void setCourseCount(Integer courseCount) {
        this.courseCount = courseCount;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }
}

File: re-ssim-service/src/main/java/com/shinonometn/re/ssim/service/courses/plugin/structure/TermMeta.kt
package com.shinonometn.re.ssim.service.courses.plugin.structure

import com.fasterxml.jackson.annotation.JsonInclude
import com.shinonometn.re.ssim.service.terms.TermInfoEntity
import org.springframework.beans.BeanUtils

@JsonInclude(JsonInclude.Include.NON_NULL)
class TermMeta {
    var code: String? = null
    var name: String? = null

    var courseCount: Int = 0
    var courseTypes: MutableList<String>? = null

    var minWeek: Int? = null
    var maxWeek: Int? = null

    var dataVersion: String? = null

    companion object {
        fun fromEntity(termInfoEntity: TermInfoEntity) = TermMeta().apply {
            BeanUtils.copyProperties(termInfoEntity, this, TermMeta::class.java)
        }
    }
}
